configure.py

```diff
@@ -1487,7 +1487,7 @@ def find_ninja():
     "test/boost/linearizing_input_stream_test.cc",
     "test/lib/log.cc",
 ]
-deps['test/boost/expr_test'] = ['test/boost/expr_test.cc', 'test/lib/expr_test_utils.cc'] + scylla_core
```

Comment:
WRT the motivation of this commit, if it fixes a bug which previous commit introduced, it's better to merge these 2, so that every commit can build and run (successful) tests.